src/Stratis.Features.SystemContracts/Identifier.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Linq;
+using NBitcoin;
+
+namespace Stratis.Features.SystemContracts
+{
+    /// <summary>
+    /// Defines the identity of a system contract. Allows the identity scheme to be flexible.
+    /// Using a ValueType is desirable here because it be used as a dictionary key.
+    /// </summary>
+    public struct Identifier
```

Comment:
Just use a `uint160` or `EmbeddedContractIdentifier`. Both would include the version so you won't need to pass that separately.

Reply:
EmbeddedContractIdentifier didn't even exist when I wrote this

Reply:
Its just the renamed SystemContractIdentifier which did exist.